allensdk/brain_observatory/behavior/data_objects/stimuli/presentations.py

```diff
@@ -540,17 +542,37 @@ def _postprocess(
                     )
                 }
             )
-        # Check if the first entry in the DataFrame is an omitted stimulus.
-        # This shouldn't happen and likely reflects some sort of camstim error
-        # with appending frames to the omitted flash frame log. See
-        # explanation here:
-        # https://github.com/AllenInstitute/AllenSDK/issues/2577
-        if "omitted" in df.columns and len(df) > 0:
-            first_row = df.iloc[0]
+        df = cls._check_for_errant_omitted_stimulus(input_df=df)
+        return df
+
+    @staticmethod
+    def _check_for_errant_omitted_stimulus(
+        input_df: pd.DataFrame
+    ) -> pd.DataFrame:
+        """Check if the first entry in the DataFrame is an omitted stimulus.
+
+        This shouldn't happen and likely reflects some sort of camstim error
+        with appending frames to the omitted flash frame log. See
+        explanation here:
+        https://github.com/AllenInstitute/AllenSDK/issues/2577
+
+        Parameters
+        ----------/
+        input_df : DataFrame
+            Input stimulus table to check for "omitted" stimulus.
+
+        Returns
+        -------
+        modified_df : DataFrame
+            Dataframe with omitted stimulus removed from first row or if not
+            found, return input_df unmodified.
+        """
+        if "omitted" in input_df.columns and len(input_df) > 0:
+            first_row = input_df.iloc[0]
             if not pd.isna(first_row["omitted"]):
```

Comment:
Is there any value for first_row{"omitted"] such that `pd.isna(first_row["omitted"]) is True` and `if first_row["omitted"] is False`?

Edit: 0 would qualify. Is there any reason with a value of 0 but not pd.isna is dropped?

Reply:
Omitted can be nan for certain sessions. This is basically just a cautions test as the behaviors can be different for numpy and pandas NA/nan values (pd.NA returns errors as ambiguous and `if np.nan` evaluates as true). If the omitted value is nan we can continue as it's not "True".